integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/DataTypeConverterUtil.scala

```diff
@@ -56,7 +56,9 @@ object DataTypeConverterUtil {
       case "decimaltype" => DataType.DECIMAL
       case "timestamptype" => DataType.TIMESTAMP
       case "datetype" => DataType.DATE
-      case "arraytype" => DataType.ARRAY
+      case "arraytype" | "arraytype(stringtype,true)" | "arraytype(inttype,true)" |
```

Comment:
I think it would be the same case with structtype as well. Please simplify as follows.
```
def convertToCarbonTypeForSpark2(dataType: String): DataType = {
    dataType.toLowerCase match {
      case "stringtype" => DataType.STRING
      case "inttype" => DataType.INT
      case "integertype" => DataType.INT
      case "tinyinttype" => DataType.SHORT
      case "shorttype" => DataType.SHORT
      case "longtype" => DataType.LONG
      case "biginttype" => DataType.LONG
      case "numerictype" => DataType.DOUBLE
      case "doubletype" => DataType.DOUBLE
      case "floattype" => DataType.DOUBLE
      case "decimaltype" => DataType.DECIMAL
      case "timestamptype" => DataType.TIMESTAMP
      case "datetype" => DataType.DATE
      case others => 
        if (others != null && others.startsWith("arraytype")) {
          DataType.ARRAY
        } else if (others != null && others.startsWith("structtype")) {
          DataType.STRUCT
        } else {
          sys.error(s"Unsupported data type: $dataType")
        }
    }
  }
```